.github/workflows/ci.yml

```diff
@@ -531,3 +549,86 @@ jobs:
           path: tests/test_docs/diffs/*.jpeg
         # Always run this step to publish test results, even on failures
         if: ${{ always() }}
+
+  end-to-end-test:
+    name: "Sandbox upgrade E2E test"
+    runs-on: ubuntu-latest
+    needs:
+      - build-new-container-image
+      - build-container-image
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install mazette tool
+        run: |
+          sudo apt install pipx
+          pipx install poetry
+          poetry install
+
+      - name: Cache mazette assets
+        id: cache-mazette
+        uses: actions/cache@v4
+        with:
+          path: |
+            share/tessdata/
+            share/vendor/
+          key: v1-mazette-linux-${{ hashFiles('./mazette.lock') }}
+
+      - name: Install mazette assets
+        if: steps.cache-mazette.outputs.cache-hit != 'true'
+        run: poetry run mazette install
+
+      - name: Login to GHCR
+        run: |
+          echo ${{ github.token }} | podman login ghcr.io -u USERNAME --password-stdin
+
+      - name: Restore container image
+        uses: actions/cache/restore@v4
+        with:
+          path: |-
+            share/container.tar
+            share/freedomofpress-dangerzone.pub
+            share/image-name.txt
+          enableCrossOsArchive: true
+          fail-on-cache-miss: true
+          key: v6-container-${{ needs.build-container-image.outputs.image_uri }}
+
+      - name: Run a conversion
+        run: |
+          poetry run ./dev_scripts/dangerzone-cli ./tests/test_docs/sample-pdf.pdf
+
+      - name: Update the settings so they check for updates
+        run: |
+          jq '.updater_check_all = true' ~/.config/dangerzone/settings.json > temp.json && mv temp.json ~/.config/dangerzone/settings.json
+
+      - name: Check the digest of the loaded podman image
+        run: |
+          podman images $(cat share/image-name.txt | head -1) --format "{{.Digest}}"
+
+      - uses: imjasonh/setup-crane@v0.1
```

Comment:
I'm about to create a `mazette.toml` under `dev_scripts/` to handle tooling like this, instead of relying to third-party GitHub actions. No need to do something now, just an FYI.